engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -1235,4 +1265,31 @@ function updateLastAction($guid, $posted = null) {
 			return false;
 		}
 	}
-}
\ No newline at end of file
+
+	/**
+	 * Get a user by GUID even if the entity is hidden or disabled
+	 *
+	 * @param int $guid User GUID. Default is logged in user
+	 *
+	 * @return \ElggUser|false
+	 * @throws UserFetchFailureException
+	 * @access private
+	 */
+	public function getUserForPermissionsCheck($guid = 0) {
+		if (!$guid) {
+			return _elgg_services()->session->getLoggedInUser();
```

Comment:
I've been wondering whether we should start discouraging the use of permission checks without passing in the user. It would make checks safer (forces devs to think what exactly they're doing instead of relying on magic) and also would reduce the complexity of core (less stuff to inject).

This is just a hunch however. I'm not sure if it would also introduce problems.